proposals/collection-expression-arguments.md

```diff
@@ -268,3 +268,59 @@ strongly triggers the view that this is simply an implicit-object-creation.  And
 case where a constructor *is* actually called (like for `Dictionary<,>`) it is misleading when calling a *create
 method*, or creating an interface.  Finally, there is general apprehension around using `new` at all as there
 is a feeling of redundancy around both the `new` indicating a new instance, *and* `[...]` indicating a new instance.
+
+## Conversions
+
+Collection arguments are *not* considered when determining *collection expression* conversions.
+
+## Construction
+
+Construction is updated as follows.
+
+The elements of a collection expression are evaluated in order, left to right.
+Within *collection arguments*, the arguments are evaluated in order, left to right.
+Each element or argument is evaluated exactly once, and any further references refer to the results of this initial evaluation.
+
+If *collection_arguments* is not the first element in the collection expression, a compile-time error is reported.
+
+If the target type is a *struct* or *class type* that implements `System.Collections.IEnumerable`, and the target type does not have a *create method*, and the target type is not a *generic parameter type* then:
+* [*Overload resolution*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1264-overload-resolution) is used to determine the best instance constructor from the *argument list*.
+  * If the *argument list* contains any values with *dynamic* type, the best instance constructor is determined at runtime.
+* If a best instance constructor is found, the constructor is invoked with the *argument list*.
+  * If the constructor has a `params` parameter, the invocation may be in expanded form.
+* Otherwise, a binding error is reported.
+
+If the target type is a type with a *create method*, then:
+* The *argument list* is a concatenation of *some `ReadOnlySpan<T>` value (TBD)* and any explicit *argument list*.
+* [*Overload resolution*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1264-overload-resolution) is used to determine the best factory method from the *argument list*.
+  * If the *argument list* contains any values with *dynamic* type, the best factory method is determined at runtime.
+* If a best factory method is found, the method is invoked with the *argument list*.
+  * If the constructor has a `params` parameter, the invocation may be in expanded form.
+* Otherwise, a binding error is reported.
+
+If the target type is an *interface type*, then:
+* [*Overload resolution*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1264-overload-resolution) is used to determine the best instance constructor from the *argument list* from the following candidate signatures:
+  * If the target type is `ICollection<E>`, or `IList<E>`, the candidates are:
```

Comment:
i would prefer we say that it's the constructors of List`1.  Basically, we already guarantee we use that type for the mutable list collections type.  So it's fine to then say that it binds to the constructors of that type (not a fixed list in the spec).

Reply:
Yeah, and on runtimes where the List<T> that is found doesn't have an `int capacity` ctor, then there are no issues.

Reply:
Same for `Dictionary<,>`.

Reply:
It looks like the equivalent constructors for `List<T>` and `Dictionary<TKey, TValue>` are available on all platforms since .NET Framework 2.0.

Reply:
my general point is simply that i don't want us to define teh set.  we should just say that it maps to the accessible (or public) constructors of that type.  :)   that way you can hit other constructors if hte runtime ever adds them.

Reply:
I'd like to leave these as explicit sets of overloads for now, and add an open question of whether the overloads should be directly taken from `List<T>` and `Dictionary<TKey, TValue>`, and if so, how that fits with the overloads for `IReadOnlyDictionary<TKey, TValue>`.

Reply:
I would say IReadOnlyList and IReadOnlyDictionary are entirely disconnected from constructors on `List<T>` and `Dictionary<TKey, TValue>` because the compiler does not guarantee that it will create a `List<T>` or a `Dictionary<TKey, TValue>` in the process of building a collection targeting a readonly interface.

Reply:
It seems unfortunate if the overloads for mutable vs. immutable interface types are spec'ed differently: that is, if the overloads for mutable interfaces are constructors of known types but the immutable interfaces are a fixed set.

Reply:
i'm genuinely ok with that.  given that the mutable interfaces are backed by real types, but the spec says the immutable ones have synthesized types. s ince we are synthesizing the type, we can synthesize the constructor called :)

Reply:
To me it seems preferable to spec them differently because that matches the bones of the underlying design, where we specced mutable to known types and readonly to compiler-generated implementations.

Reply:
I would want it to follow along, too. If the spec started promising a given concrete implementation for a readonly interface, then we would begin tracking the constructors of that promised concrete collection type for that readonly interface.

Reply:
strong agree with Joseph.